src/ArduinoIoTCloudTCP.cpp

```diff
@@ -69,6 +69,11 @@ extern "C" void updateTimezoneInfo()
   ArduinoCloud.updateInternalTimezoneInfo();
 }
 
+extern "C" void setThingIdOutdated()
```

Comment:
Where is this function called? Why is there an `extern "C"` declaration necessary? (Looks like it's passed to C code as a callback function).

Reply:
This function is used as callback function fot the update event of the `_thing_id` property. Each time a new thing_id value is received this function is called to restart a new configuration process. It should not be necessary, i think i've copy pasted from 

`extern "C" void updateTimezoneInfo()`

that i have again copy pasted from

`extern "C" unsigned long getTime()`

but i think it is safe to remove all three `extern "C"` declarations

Reply:
Allright, but where is this function registered as a callback?

Reply:
Here https://github.com/arduino-libraries/ArduinoIoTCloud/pull/297/commits/db3e9433c612998ddf0a0c3e9f3027a446d9c588 i think i've messed up a bit rebasing this firsts commits. I'll fix it keeping 

`  addPropertyReal(_thing_id, _device_property_container, "thing_id", Permission::ReadWrite)` in the first and adding

`onUpdate(setThingIdOutdated);` in the second one

Reply:
I've also added a new commit to remove the extern `"C" declaration` see https://github.com/arduino-libraries/ArduinoIoTCloud/pull/297/commits/0541f7bbd19214c4a1402d20631786ff4b8e722c